online_party_system/party_match.go

```diff
@@ -129,19 +130,27 @@ func getBlockedFriendsWithCache(ctx context.Context, logger runtime.Logger, nk r
 	friendState := 3 // 3 is Friend_BLOCKED
 	cursor := ""     // no cursor
 
-	friendsPtrArray, cursor, err := nk.FriendsList(ctx, userID, friendLimit, &friendState, cursor)
-	if err != nil {
-		logger.Error("Error getting FriendsList: %v", err)
-		return nil
-	}
+	var blockedFriends []string
 
-	blockedFriends := make([]string, 0, len(friendsPtrArray))
+	for {
+		var friendsPtrArray []*api.Friend
+		var err error
+		friendsPtrArray, cursor, err = nk.FriendsList(ctx, userID, friendLimit, &friendState, cursor)
+		if err != nil {
+			logger.Error("Error getting FriendsList: %v", err)
+			return nil
+		}
+
+		for _, friendPtr := range friendsPtrArray {
+			friend := *friendPtr
+			user := friend.User
+			userID := user.Id
+			blockedFriends = append(blockedFriends, userID)
```

Comment:
This can just be collapsed to something like:
```
for _, friend := range friendsPtrArray {
    blockedFriends = append(blockedFriends, friend.User.Id)
}
```